### ISSUE TITLE
Piecewise bug with scalar domains and more than 2 conditions

### ISSUE THREAD

**Issue by jakobgager** [numpy/numpy#5729]
(Commented in #4792)
There, seems to be an issue with scalar domains and three or more conditions:
```python
>>> np.piecewise(3, [True, False, False],[4,2,0])
ValueError: function list and condition list must be the same    
```
Funny enough it works fine with two conditions (this is what has been tested):
```python
>>> np.piecewise(3, [True, False],[4,2])
array(4)
```

The three conditions version works fine with _select_:
```python
>>> np.select([True, False, False],[4,2,0])
array(4)
``` 
Tested on python 3.4 with numpy 1.9.0 and python 2.7.6 with numpy 1.8.2

**Comment by mforbes** [comment]
Confirmed.

**Comment by WarrenWeckesser** [comment]
This appears to be fixed.  Close it?

**Comment by mforbes** [comment]
Seems to be fixed for me.

**Comment by bastula** [comment]
I guess I never updated this issue with a comment? It should be resolved by PR #7145.

**Comment by charris** [comment]
Closing this then. @WarrenWeckesser Thanks for the heads up.
